pkgs/development/java-modules/maven-fod.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, stdenv
+, maven
+}:
+
+{ src
+, patches ? []
+, pname
+, version
+, mvnSha256 ? "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
+, mvnHash ? "sha256-${mvnSha256}"
+, mvnFetchExtraArgs ? {}
+, ...
+} @args:
+
+# originally extracted from dbeaver
+# created to allow using maven packages in the same style as rust
+
+stdenv.mkDerivation (rec {
+  fetchedMavenDeps = stdenv.mkDerivation ({
+    name = "${pname}-${version}-maven-deps";
+    inherit src;
+
+    buildInputs = [
+      maven
+    ];
+
+    buildPhase = ''
+      mvn package -Dmaven.repo.local=$out/.m2 -P desktop,all-platforms
```

Comment:
Maybe we should make the list of profiles (`desktop,all-platforms` in the case of `dbeaver`) configurable? Same for the `buildPhase`.

I'm not entirely sure if maven cares if a profile does not exist. At least a warning is logged when trying to build a non-existing profile (in this case `scim-for-keycloak`):

```
[WARNING] The requested profile "desktop" could not be activated because it does not exist.
[WARNING] The requested profile "all-platforms" could not be activated because it does not exist.
```

Reply:
Added `mvnParameters`